packages/flutter/test/material/app_bar_test.dart

```diff
@@ -903,30 +903,92 @@ void main() {
     expect(tabBarHeight(tester), initialTabBarHeight);
   });
 
+  testWidgets('AppBar uses the specified elevation or defaults to 4.0', (WidgetTester tester) async {
+    Widget buildAppBar([double? elevation]) {
+      return MaterialApp(
+        home: Scaffold(
+          appBar: AppBar(title: const Text('Title'), elevation: elevation),
+        ),
+      );
+    }
+
+    Material getMaterial() => tester.widget<Material>(find.descendant(
+      of: find.byType(AppBar),
+      matching: find.byType(Material),
+    ));
+
+    // Default elevation should be _AppBarState._defaultElevation = 4.0
+    await tester.pumpWidget(buildAppBar());
+    expect(getMaterial().elevation, 4.0);
+
+    // AppBar should use the specified elevation.
+    await tester.pumpWidget(buildAppBar(8.0));
+    expect(getMaterial().elevation, 8.0);
+  });
+
   testWidgets('SliverAppBar rebuilds when forceElevated changes', (WidgetTester tester) async {
     // Regression test for https://github.com/flutter/flutter/issues/59158.
-    Widget buildSliverAppBar(bool forceElevated) {
+    Widget buildSliverAppBar(bool forceElevated, [double? elevation]) {
       return MaterialApp(
         home: CustomScrollView(
           slivers: <Widget>[
             SliverAppBar(
               backwardsCompatibility: false,
               title: const Text('Title'),
               forceElevated: forceElevated,
+              elevation: elevation,
             ),
           ],
         ),
       );
     }
 
-    final Finder appBarFinder = find.byType(AppBar);
-    AppBar getAppBarWidget(Finder finder) => tester.widget<AppBar>(finder);
+    AppBar getAppBar() => tester.widget<AppBar>(find.byType(AppBar));
+    Material getMaterial() => tester.widget<Material>(find.byType(Material));
 
+    // When forceElevated is off, SliverAppBar should not be elevated.
     await tester.pumpWidget(buildSliverAppBar(false));
-    expect(getAppBarWidget(appBarFinder).elevation, 0.0);
+    expect(getMaterial().elevation, 0.0);
 
+    // Default elevation should be _AppBarState._defaultElevation = 4.0, and
+    // the AppBar's elevation should not be specified by SliverAppBar.
     await tester.pumpWidget(buildSliverAppBar(true));
-    expect(getAppBarWidget(appBarFinder).elevation, 4.0);
+    expect(getMaterial().elevation, 4.0);
+    expect(getAppBar().elevation, null);
+
+    // SliverAppBar should use the specified elevation.
+    await tester.pumpWidget(buildSliverAppBar(true, 8.0));
+    expect(getMaterial().elevation, 8.0);
+  });
+
+  testWidgets('SliverAppBar uses elevation of AppBarTheme', (WidgetTester tester) async {
+    // Regression test for https://github.com/flutter/flutter/issues/73525.
+    const double themeElevation = 12.0;
+
+    Widget buildSliverAppBar(bool forceElevated, double elevation) {
```

Comment:
The common methods here could be broken out and shared to reduce boilerplate. You can `group` common tests to share methods like this. Example: https://github.com/flutter/flutter/blob/a2a77285dceaed7fcc284f8c0abfdb511f1c545e/packages/flutter/test/material/floating_action_button_location_test.dart#L611